src/filters/path.rs

```diff
@@ -186,15 +186,19 @@ pub fn index() -> impl Filter<Extract=(), Error=Rejection> + Copy {
 ///         format!("You asked for /{}", id)
 ///     });
 /// ```
-pub fn param<T: FromStr + Send>() -> impl Filter<Extract=One<T>, Error=Rejection> + Copy {
+pub fn param<T>() -> impl Filter<Extract=One<T>, Error=Rejection> + Copy
+where
+    T: FromStr + Send,
+    T::Err: Into<::reject::Cause>
```

Comment:
This would be a breaking change, though it definitely seems worth doing as part of 0.2!